backend/src/modules/transactions/transactions.controller.ts

```diff
@@ -23,9 +27,12 @@ import { TransactionQueryService } from './service/transaction-query.service';
 export class TransactionsController {
   constructor(
     private readonly transactionsService: TransactionsService,
+    private readonly categoryService: CategoryService,
     private readonly queryService: TransactionQueryService,
     private readonly logger: Logger,
-  ) {}
+  ) {
+    this.categoryService.loadCategories();
```

Comment:
As tik dabar supratau, kad tu seedini categories duombazes duomenis `TransactionController`. Tai yra negerai del keleto dalyku pvz.: nereikalingas performance hit ( tu atlieki operacijas kurias reikia atlikti **viena karta** per visa aplikacijos gyvavimo laika kiekviena karta kai transactionControlleris yra sukuriamas), nebegalima buna istrinti default kategoriju, neiseina priskirti atskiru kategoriju atskiriem useriam ir t.t.

Sprendimas kurio tu ieskai sitai problemai yra vadinamas `Database seeding`. Look it up

Reply:
Be to sitoj vietoj reiketu apsispresti ar reikia is viso reikia db dabar. Jei darom be db, tai viskas labai paprastai, categoryService tures savyje ta hardocdinta masyva, ir tokie dalykai kaaip findAll categories tiesiog grazins ta masyva. Zymiai paprasciau ir tai yra pusiau reliable variantas,  jeigu nenorim tvarkingai seedint duombazes.

Reply:
Bet tada nesigaus, kad nera kaip prideti nauju kategoriju ir kaip editinti bazines? O jeigu seedint db tai cia turetu buti kazkoks visiskai atskiras skriptas?

Reply:
Jo, jei be db darom tai negalim editint ir kurt nauju kategoriju. Tai in the long run, butu geriau tureti db bet reikia teisingai padaryti.
O seedinimas tia jo, butu atskiras kazkoks scriptas/toolas, kuri cj reiketu parunnint viena karta pries pirma karta leidziant backa, koks nors `npm run seed`

https://www.google.lt/search?ei=y1rNXMKLA-qymwXn0pPoCA&q=seeding+mong+odb
https://www.npmjs.com/package/mongoose-seed
https://docs.mongodb.com/manual/reference/program/mongoimport/

Cia kazkokie linkai, reikia daugiau pasiskaityt

Reply:
Cj butu gerai taip padaryt kaip sitas vyrukas daro:
https://www.youtube.com/watch?v=V30Rpqi6kYE
Visus tuos hardcodintus kur dabar turi kontroleryje taip i seed faila ir tuomet padarai `package.json` scripta `seed: node seeds/categories.js` or something kas paleistu ir viskas bus ciki